app/controllers/SupplementaryUnitController.scala

```diff
@@ -53,15 +54,31 @@ class SupplementaryUnitController @Inject() (
         case Some(value) => form.fill(value)
       }
 
-      Ok(view(preparedForm, mode, recordId))
+      request.userAnswers.get(RecordCategorisationsQuery) match {
+        case None        => Redirect(routes.JourneyRecoveryController.onPageLoad().url)
+        case Some(query) =>
+          query.records(recordId).measurementUnit match {
+            case None                  => Ok(view(preparedForm, mode, recordId, ""))
+            case Some(measurementUnit) => Ok(view(preparedForm, mode, recordId, measurementUnit))
+          }
+      }
   }
 
   def onSubmit(mode: Mode, recordId: String): Action[AnyContent] =
     (identify andThen getData andThen requireData).async { implicit request =>
       form
         .bindFromRequest()
         .fold(
-          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, recordId))),
+          formWithErrors =>
```

Comment:
Some formatting stuff happens with big numbers

Enter 1234567890.123456
![image](https://github.com/hmrc/trader-goods-profiles-frontend/assets/114592674/7024bcfa-3400-45d4-92ac-3d529db267ba)

It turns it into scientific notation
![image](https://github.com/hmrc/trader-goods-profiles-frontend/assets/114592674/0a48e73a-9004-420d-891f-0bce8150a758)

Click Change

it is still in scientific notation
![image](https://github.com/hmrc/trader-goods-profiles-frontend/assets/114592674/e61e3cd9-9d85-4bb6-8b3c-e06c58aa80d1)

Without changing the value, if I click continue its unhappy with the format
![image](https://github.com/hmrc/trader-goods-profiles-frontend/assets/114592674/4cc36c26-8ba6-48f4-aeb7-155466d04279)

Should probably ensure its displayed in full form everytime?

Reply:
This is now fixed. I have changed formProvider from double to String